examples/usekeypress/usekeypress.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+const {useState} = require('react');
+const React = require('react');
+const {render, useKeypress} = require('../..');
+
+const {exit} = process;
+
+const Robot = () => {
+	const [x, setX] = useState(1);
+	const [y, setY] = useState(1);
+
+	useKeypress(str => {
```

Comment:
```suggestion
	useKeypress(key => {
```

Reply:
so useKeypress calls the callback with the string representation of the key, plus the Key object `(str, key)`. In this case, we just care about the string representation. Its might be a bit confusing and we should just choose the full Key object representation, allowing people to destructure with `({ str })` perhaps. What are your thoughts

Reply:
What are the use cases for wanting the full object, not just the string representation?

Reply:
if `ctrl` is held down etc I'd guess?

Reply:
It could be passed as a second argument to the callback in that case:

```js
useKeypress((key, { ctrl }) => ...)
```

Reply:
That's what happens with the current implementation

Reply:
key object is probably more useful, btw.

```js
const readline = require('readline');
readline.emitKeypressEvents(process.stdin);
process.stdin.setRawMode(true);
process.stdin.on('keypress', console.log);
```

```sh-session
$ node script.js
# hit s
s { sequence: 's',
  name: 's',
  ctrl: false,
  meta: false,
  shift: false }
# hit shift + s
S { sequence: 'S', name: 's', ctrl: false, meta: false, shift: true }
# hit ctrl + s
 { sequence: '\u0013',
  name: 's',
  ctrl: true,
  meta: false,
  shift: false }
```


So I think just a single `Key` object might be better?

Reply:
I'd keep first argument as a key string for convenience to avoid destructuring when you only care about what key was pressed and keep `Key` object as a 2nd argument with `ctrl`, `shift`, etc props.

Reply:
We could experiment with alternative, simpler apis and how they feel with higher level hooks maybe like ‘useKey = cb => useKeypress((_, key) => cb(key))’

Other ways to interpret the key object into a string might be a good middle ground as well.

‘useKeystring = cb => useKeypress((_, key) => cb(keyToString(key)))’

Where keyToString converts the key object to strings like ‘meta+shift+s’ or ‘meta+S’.